Library/Homebrew/formula_auditor.rb

```diff
@@ -865,14 +872,29 @@ def head_only?(formula)
     end
 
     def linux_only_gcc_dep?(formula)
-      # TODO: Make this check work when running on Linux and not simulating macOS too.
-      return false unless Homebrew::SimulateSystem.simulating_or_running_on_macos?
+      odie "`#linux_only_gcc_dep?` works only on Linux!" if Homebrew::SimulateSystem.simulating_or_running_on_macos?
 
       formula_hash = formula.to_hash_with_variations
-      deps = formula_hash["dependencies"]
-      linux_deps = formula_hash.dig("variations", :x86_64_linux, "dependencies")
+      linux_deps = formula_hash["dependencies"]
+      return false if linux_deps.exclude?("gcc")
+
+      variations = formula_hash["variations"]
+      # The formula has no variations, so all versions depend on GCC.
+      return false if variations.blank?
+
+      # FIXME: This returns a false positive for formulae that do, for example:
+      # ```ruby
+      # on_system :linux, macos: :catalina_or_newer do
+      #   depends_on "gcc"
+      # end
+      # ```
+      variations_deps = []
+      formula_hash["variations"].each_value do |data|
```

Comment:
I guess what I need to do is to check if all possible variations appear as keys in `formula_hash["variations"]`. If one is missing, then that means that variation has identical data to `:x86_64_linux`.